### DISCUSSION TITLE
Examples - how to run them from CLI (newbie)

### DISCUSSION THREAD

**Discussion by anrl** [eclipse-langium/langium#1768] [Q&A] [answered]
Hi

I am trying to run the arithmetic example. I did the npm install from the root of the project and also the 
arithmetic example directory. I still don't have the ../out/cli/cli.js created. Am I missing some setup?
Any help would be deeply appreciated.

**Comment by JohannesMeierSE** [comment, marked as answer]
Hello @anrl,

thanks for your feedback!

For me, the following worked:
```
git clone ...
cd langium
npm install
npm run build
cd examples/arithmetics
node ./bin/cli eval example/example.calc
```

... and produced this result:

```
line 14: 2 * c ===> 16
line 15: b % 2 ===> 1
line 18: Root(D, 3) ===> 4.999999999999999
line 19: Root(64, 3) ===> 3.9999999999999996
line 20: Sqrt(81) ===> 9
```

Could you try this on your side?
Then I would create a PR for updating the documentation!

**Reply by msujew** [reply]
@anrl I believe there are two steps that are missing/incorrect that are correctly mentioned in the post by @JohannesMeierSE:

1. You only need to run the `npm install` once in the root directory. We use npm workspaces to manage all of our npm projects as one monorepo.
2. You need to run `npm run build` to actually generate the JavaScript code from our TypeScript sources.

We have a [contributing guide](https://github.com/eclipse-langium/langium/blob/main/CONTRIBUTING.md#developing-langium) that also explains how to build/contribute to this repo.

**Comment by anrl** [comment]
Thanks a lot!
`npm install`
`npm run build`
at the root of langium directory did it. Now, I am able to run the examples!

**Reply by JohannesMeierSE** [reply]
Thanks for your feedback, I fixed this issue in the documentation for all four examples: https://github.com/eclipse-langium/langium/pull/1889
